### PR TITLE
Added Java Solution for tracks

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

**Comment by HarryWangATX** [comment]
This solution does TLE on the judge. I was not able to find any java solution that passed eihter. Maybe you guys might have a way to solve this, but I think it is the oj.uz's constraints.

File: content/4_Gold/BFS.mdx [hunk 1]
```diff
@@ -576,6 +576,134 @@ int main() {
...
+                    count[nx][ny] = count[curr[0]][curr[1]];
+                    q.addFirst(new int[]{nx, ny});
+                }
+            }
+        }
+
+        return N;
+    }
+
+    private static class FastIO extends PrintWriter {
```
Comment (SansPapyrus683) [inline content/4_Gold/BFS.mdx:642]: just asking, is fastio 100% needed for this? if not, then it's probably best not to add it (or at least wrap it in a snippet)?

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:642]: It will definitely TLE without it bc it TLEs with it. Ig I'll wrap it in a snippet.

Comment (SansPapyrus683) [reply content/4_Gold/BFS.mdx:642]: oh yeah sorry i didn't see the first comment

File: content/4_Gold/BFS.mdx [hunk 2]
```diff
@@ -576,6 +576,135 @@ int main() {
...
+        }
+        public double nextDouble() { return Double.parseDouble(next()); }
+    }
+	//EndCodeSnip
+}
+```
+
+<Warning>
+
+Due to oj.uz's grading constraints for Java, this solution will TLE on the judge.
```
Comment (dolphingarlic) [inline content/4_Gold/BFS.mdx:702]: Are there no constant optimizations you can do to prevent this TLE?

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:702]: I tried, but needs 100ms rn.

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:702]: also checked oj.uz, there were no Java ACs

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
use tabs

File: content/4_Gold/BFS.mdx (hunk 2 above)
Comment (dolphingarlic) [reply content/4_Gold/BFS.mdx:702]: Maybe try CSES and if it passes there, just pretend oj.uz doesn't exist

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:702]: can u provide a link?

Comment (dongliu0426) [reply content/4_Gold/BFS.mdx:702]: @HarryWangATX https://cses.fi/104/list/

Comment (dongliu0426) [reply content/4_Gold/BFS.mdx:702]: TLEs as well on cses

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:702]: hmmmmmmmm. do u guys see any optimization spots?

Comment (dongliu0426) [reply content/4_Gold/BFS.mdx:702]: maybe try changing `int[]` to a custom `Pair` in the queue?

Comment (HarryWangATX) [reply content/4_Gold/BFS.mdx:702]: still TLEs on the same testcases

**Review by dongliu0426 — APPROVED** [review]
looks good. idk about merging this (bc it TLEs)

**Comment by thecodingwizard** [comment]
I'm just going to merge. @HarryWangATX can you also add a note to the BFS module where the problem is to note that Java users may TLE?

**Comment by HarryWangATX** [comment]
> I'm just going to merge. @HarryWangATX can you also add a note to the BFS module where the problem is to note that Java users may TLE?

Yep, already added in this pr.

**Pull request merged**